src/object.rs

```diff
@@ -0,0 +1,334 @@
+
+            let symbol_code = b[35] as char;
+
+            let comment = b[36..].to_owned();
+
+            Ok(Self {
+                to,
+                name,
+                live,
+                timestamp,
+                latitude,
+                longitude,
+                precision,
+                symbol_table,
+                symbol_code,
+                comment,
+                cst: AprsCst::Uncompressed,
+            })
+        }
+    }
+
+    pub fn encode<W: Write>(&self, buf: &mut W) -> Result<(), EncodeError> {
+        // format for uncompressed is
+        // N Bytes      Description/Value
+        // -----------------------------
+        // [9]          (object name, space-padded)
+        // [1]          (object live =  '*' dead = '_')
+        // [7]          (timestamp)
+        // [8]          (latitude)
+        // [1]          (sym table id)
+        // [9]          (longitude)
+        // [1]          (sym code)
+        // [7/0]        (extended data [NOT PARSING])
+        // [0-36/0-43]  (comment)
+        //
+
+        // format for compressed is
+        // N Bytes      Description/Value
+        // -----------------------------
+        // [9]          (object name, space-padded)
+        // [1]          (object live =  '*' dead = '_')
+        // [7]          (timestamp)
+        // [13]         (compressed position data)
+        // [43]         (comment)
+
+        let mut name = self.name.clone();
+        name.truncate(9);
+        write!(buf, ";")?;
+        write!(buf, "{name:9}")?;
+        write!(buf, "{}", if self.live { '*' } else { ' ' })?;
+        self.timestamp.encode(buf)?;
+        match self.cst {
+            AprsCst::CompressedSome { cs, t } => {
+                write!(buf, "{}", self.symbol_table)?;
+
+                self.latitude.encode_compressed(buf)?;
+                self.longitude.encode_compressed(buf)?;
+
+                write!(buf, "{}", self.symbol_code)?;
+
+                cs.encode(buf, t)?;
+            }
+            AprsCst::CompressedNone => {
+                write!(buf, "{}", self.symbol_table)?;
+
+                self.latitude.encode_compressed(buf)?;
+                self.longitude.encode_compressed(buf)?;
+
+                write!(buf, "{}", self.symbol_code)?;
+
+                write!(buf, " sT")?; // no cst
+            }
+            AprsCst::Uncompressed => {
+                self.latitude.encode_uncompressed(buf, self.precision)?;
+                write!(buf, "{}", self.symbol_table)?;
+                self.longitude.encode_uncompressed(buf)?;
+                write!(buf, "{}", self.symbol_code)?;
+            }
+        }
+
+        buf.write_all(&self.comment)?;
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+
+    use crate::{AprsData, AprsPacket};
+
+    use super::*;
+
+    #[test]
+    fn parse_uncompressed_live_object() -> Result<(), Box<dyn std::error::Error>> {
```

Comment:
I've never seen `-> Result<...>` used on a `test` function - neat. However I think I prefer just using `unwrap` where required - it makes it more obvious exactly which line failed.